homeassistant/components/efesto/climate.py

```diff
@@ -0,0 +1,213 @@
+"""Support for Efesto heating devices."""
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
+from homeassistant.components.climate.const import (
+    SUPPORT_ON_OFF, ATTR_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_URL, CONF_PASSWORD, CONF_USERNAME, CONF_DEVICE,
+    PRECISION_WHOLE, STATE_OFF, STATE_ON, TEMP_CELSIUS
+)
+
+REQUIREMENTS = ['efestoclient==0.0.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_DEVICE): cv.string,
+})
+
+ATTR_DEVICE_STATUS = 'device_status'
+ATTR_SMOKE_TEMP = 'smoke_temperature'
+ATTR_REAL_POWER = 'real_power'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Efesto device."""
+    url = config.get(CONF_URL)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    device = config.get(CONF_DEVICE)
+
+    return _setup_efesto(
+        url, username, password, device, config, add_entities
+    )
+
+
+def _setup_efesto(url, username, password, device, config, add_entities):
+    """Set up the rounding function."""
+    from efestoclient import EfestoClient
+
+    client = EfestoClient(url, username, password, device, False)
+
+    add_entities(
+        [EfestoHeatingDevice(client)],
+        True
+    )
+
+    return True
+
+
+class EfestoHeatingDevice(ClimateDevice):
+    """Representation of a Efesto heating device."""
+
+    def __init__(self, client):
+        """Initialize the thermostat."""
+        self.client = client
+        self._on = False
+        self._device_status = None
+        self._idle_info = None
+        self._operation_mode = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._smoke_temperature = None
+        self._real_power = None
+        self._name = 'Efesto'
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        supported = (
+            SUPPORT_TARGET_TEMPERATURE | SUPPORT_ON_OFF
+        )
+        return supported
+
+    @property
+    def state(self):
+        """Return the current state."""
+        if (self._operation_mode and
+                self._operation_mode not in ['OFF', 'ON']):
+            return self._operation_mode
+        if self.is_on is False:
+            return STATE_OFF
+        if self.is_on:
+            return STATE_ON
+        return None
```

Comment:
Correct but my problem with the default base class was that the 'on' state did not work correctly.
It didn't recognize that the device was 'on' when self._operation_mode was set to 'ON' (because it maps 'on' state to STATE_ON which is lowercase 'on').
So with the default base state def the resulting state was 'ON' but the on/off slider itself was switched off.
That was the main reason why I created my own state definition.

Reply:
State is calculated by base class.